polygeist/tools/cgeist/Lib/CGCall.cc

```diff
@@ -444,7 +448,7 @@ ValueCategory MLIRScanner::VisitCallExpr(clang::CallExpr *Expr) {
     llvm::dbgs() << "\n";
   });
 
-  auto Loc = getMLIRLocation(Expr->getExprLoc());
+  Location Loc = getMLIRLocation(Expr->getExprLoc());
```

Comment:
Do we need this? `getMLIRLocation` is a common function in our codebase and the name suggests it returns an `mlir::Location`.

Reply:
The LLVM coding guidelines state:

"Some are advocating a policy of “almost always auto” in C++11, however LLVM uses a more moderate stance. Use auto if and only if it makes the code more readable or easier to maintain. Don’t “almost always” use auto, but do use auto with initializers like cast<Foo>(...) or other places where the type is already obvious from the context. Another time when auto works well for these purposes is when the type would have been abstracted away anyways, often behind a container’s typedef such as std::vector<T>::iterator."

The type is not obvious here because the context is a function call and relying on the function name is technically not something that undoubtedly tell use the function return type.  Just following coding guidelines.